e2e/cloud-plugins-suite/azure-monitor.spec.ts

```diff
@@ -21,6 +21,8 @@ type AzureMonitorConfig = {
 
 type AzureMonitorProvision = { datasources: AzureMonitorConfig[] };
 
+const dataSourceName = 'Azure Monitor E2E Tests';
```

Comment:
the problem of using a fixed data source name is that if the tests are stopped in the middle, it won't work after restarting (since it won't delete it). Can you add some randomized string at the end?

Reply:
I believe if you don't specify the name line 35 the e2e test will come up with a randomized name for you

Reply:
🤔 I see you use it in `addPanel`. Could be get the name from the datasource created?
If not, adding a randomised string at the end is a good way forward too

Reply:
I'll add a randomised string at the end. I did originally try to retrieve the random name generated but because of the way the Cypress `wrap` function works this didn't seem like it was possible. The E2E scenarios are structured in such a way that all changes in a scenario are reverted at the end of the scenario which is why I went with the fixed name 😊 

I will also add the `skipScenario` flag to the annotations test and then this should be good to merge!